### PR TITLE
Prepare copyright rule for W3C Inc for 2023

### LINKED ISSUE #1681: Transition to the W3C Inc copyright starting from Jan 1, 2023
Starting from January 1, 2023, TR documents must use the following copyright:
```
<p class="copyright"><a href="https://www.w3.org/Consortium/Legal/ipr-notice#Copyright">Copyright</a> © 2023 <a href="https://www.w3.org/"> World Wide Web Consortium, Inc.</a>. <abbr title="World Wide Web Consortium">W3C</abbr><sup>®</sup> <a href="https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer">liability</a>, <a href="https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks">trademark</a> and <a href="https://www.w3.org/Consortium/Legal/copyright-software">permissive document license</a> rules apply.</p>
```

After discussion with @plehegar, the previous copyright will be allowed until January 31, 2023.

/cc @wseltzer

### PR DESCRIPTION
**Pull request by jennyliang220**
- Add a way to run single test. 
- Prepare copyright rule for W3C Inc for 2023, resolves #1681.

#### Transition rule:

- Before 2023.1.1: Use legacy copyright with 4 hosts.
- 2023.1.1 - 2023.2.1: Start accepting `W3C Inc.` copyright, give warning for the legacy copyright.
- After 2023.2.1: Accept `W3C Inc.` only.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,13 +99,49 @@ from the root and you will be running the test suite. Mocha can be installed wit
...
+
+The testcase document can run independently
+
+```bash
+$ npm run testserver
+```
+
+#### 4. Run certain test
+
+Add process env before `npm run test` and `describe.only()`to run single test.
```
Comment (deniak) [inline README.md:121]:
```suggestion
Add process env before `npm run test` and `describe.only()` to run single test.
```

File: lib/l10n.js [hunk 2]
```diff
@@ -149,10 +149,12 @@ export function message(profileCode, rule, key, extra) {
                         `[${profile}][${rule.section}][${rule.rule}].text`
                     );
                 }
+                // TODO: to change after 2023
```
Comment (deniak) [inline lib/l10n.js:152]: Please, create an issue listing everything that needs to be rolled back in January so we don't miss these TODOs

File: lib/rules.json [hunk 3]
```diff
@@ -1587,7 +1587,7 @@
...
...license</a> rules apply. </blockquote> </div><div class=\"source\"> <span style=\"font-style: italic\">Include this source code</span>:<br><code>&lt;p class=\"copyright\"&gt;&lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\"&gt;Copyright&lt;/a&gt; © 2022&lt;a href=\"https://www.w3.org/\"&gt;&lt;abbr title=\"World Wide Web Consortium\"&gt;W3C&lt;/abbr&gt;&lt;/a&gt;&lt;sup&gt;®&lt;/sup&gt; (&lt;a href=\"https://www.csail.mit.edu/\"&gt;&lt;abbr title=\"Massachusetts Institute of Technology\"&gt;MIT&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.ercim.eu/\"&gt;&lt;abbr title=\"European Research Consortium for Informatics and Mathematics\"&gt;ERCIM&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.keio.ac.jp/\"&gt;Keio&lt;/a&gt;, &lt;a href=\"https://ev.buaa.edu.cn/\"&gt;Beihang&lt;/a&gt;). W3C &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\"&gt;liability&lt;/a&gt;, &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\"&gt;trademark&lt;/a&gt; and &lt;a href=\"https://www.w3.org/Consortium/Legal/copyright-software\"&gt;permissive document license&lt;/a&gt; rules apply.&lt;/p&gt;</code> </div></li></ol> </li></ul>",
```
Comment (deniak) [inline lib/rules.json:1590]:
```suggestion
                    "copyright": "The copyright depends on the publication date:<ul><li>starting from 01 February 2023, the copyright <span class=\"rfc2119\">must</span> follow the following markup (fill in with the appropriate year, years, or year range). The type of license the document is using can be found in the group's charter. <ol> <li>For documents using W3C Document License: <blockquote class=\"copyright\"><a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\">Copyright</a> © @{year} <a href=\"https://www.w3.org/\">World Wide Web Consortium, Inc.</a>. <abbr title=\"World Wide Web Consortium\">W3C</abbr><sup>®</sup> <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\">liability</a>, <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\">trademark</a> and <a href=\"https://www.w3.org/Consortium/Legal/copyright-documents\">document use</a> rules apply.</blockquote><div class=\"source\"> <span style=\"font-style: italic\">Include this source code</span>:<br><code>&lt;p class=&quot;copyright&quot;&gt;&lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#Copyright&quot;&gt;Copyright&lt;/a&gt; © @{year} &lt;a href=&quot;https://www.w3.org/&quot;&gt;World Wide Web Consortium, Inc.&lt;/a&gt;. &lt;abbr title=&quot;World Wide Web Consortium&quot;&gt;W3C&lt;/abbr&gt;&lt;sup&gt;®&lt;/sup&gt; &lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer&quot;&gt;liability&lt;/a&gt;, &lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks&quot;&gt;trademark&lt;/a&gt; and &lt;a href=\"https://www.w3.org/Consortium/Legal/copyright-documents\"&gt;document use&lt;/a&gt; rules apply.&lt;/p&gt;</code> </div></li><li>For documents using W3C Software and Document License: <blockquote class=\"copyright\"><a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\">Copyright</a> © @{year} <a href=\"https://www.w3.org/\">World Wide Web Consortium, Inc.</a>. <abbr title=\"World Wide Web Consortium\">W3C</abbr><sup>®</sup> <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\">liability</a>, <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\">trademark</a> and <a href=\"https://www.w3.org/Consortium/Legal/copyright-software\">permissive document license</a> rules apply.</blockquote><div class=\"source\"> <span style=\"font-style: italic\">Include this source code</span>:<br><code>&lt;p class=&quot;copyright&quot;&gt;&lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#Copyright&quot;&gt;Copyright&lt;/a&gt; © @{year} &lt;a href=&quot;https://www.w3.org/&quot;&gt;World Wide Web Consortium, Inc.&lt;/a&gt;. &lt;abbr title=&quot;World Wide Web Consortium&quot;&gt;W3C&lt;/abbr&gt;&lt;sup&gt;®&lt;/sup&gt; &lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer&quot;&gt;liability&lt;/a&gt;, &lt;a href=&quot;https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks&quot;&gt;trademark&lt;/a&gt; and &lt;a href=&quot;https://www.w3.org/Consortium/Legal/copyright-software&quot;&gt;permissive document license&lt;/a&gt; rules apply.&lt;/p&gt;</code> </div></li></ol></li><li>during the transition period (between 01 January 2023 and 01 February 2023), both the new and legacy copyrights are allowed.</li><li>until 31 December 2022, use the legacy copyright following the markup below (fill in with the appropriate year, years, or year range). The type of license the document is using can be found in the group's charter. <ol> <li>For documents using W3C Document License: <div class=\"boilerplate\"> <blockquote class=\"copyright\"><a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\">Copyright</a> © 2022<a href=\"https://www.w3.org/\"> <abbr title=\"World Wide Web Consortium\">W3C</abbr> </a> <sup>®</sup> (<a href=\"https://www.csail.mit.edu/\"> <abbr title=\"Massachusetts Institute of Technology\">MIT</abbr> </a>, <a href=\"https://www.ercim.eu/\"> <abbr title=\"European Research Consortium for Informatics and Mathematics\">ERCIM</abbr> </a>, <a href=\"https://www.keio.ac.jp/\">Keio</a>, <a href=\"https://ev.buaa.edu.cn/\">Beihang</a>). W3C <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\">liability</a>, <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\">trademark</a> and <a href=\"https://www.w3.org/Consortium/Legal/copyright-documents\">document use</a> rules apply. </blockquote> </div><div class=\"source\"> <span style=\"font-style: italic\">Include this source code</span>:<br><code>&lt;p class=\"copyright\"&gt;&lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\"&gt;Copyright&lt;/a&gt; © 2022&lt;a href=\"https://www.w3.org/\"&gt;&lt;abbr title=\"World Wide Web Consortium\"&gt;W3C&lt;/abbr&gt;&lt;/a&gt;&lt;sup&gt;®&lt;/sup&gt; (&lt;a href=\"https://www.csail.mit.edu/\"&gt;&lt;abbr title=\"Massachusetts Institute of Technology\"&gt;MIT&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.ercim.eu/\"&gt;&lt;abbr title=\"European Research Consortium for Informatics and Mathematics\"&gt;ERCIM&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.keio.ac.jp/\"&gt;Keio&lt;/a&gt;, &lt;a href=\"https://ev.buaa.edu.cn/\"&gt;Beihang&lt;/a&gt;). W3C &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\"&gt;liability&lt;/a&gt;, &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\"&gt;trademark&lt;/a&gt; and &lt;a href=\"https://www.w3.org/Consortium/Legal/copyright-documents\"&gt;document use&lt;/a&gt; rules apply.&lt;/p&gt;</code> </div></li><li>For documents using W3C Software and Document License: <div class=\"boilerplate\"> <blockquote class=\"copyright\"><a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\">Copyright</a> © 2022<a href=\"https://www.w3.org/\"> <abbr title=\"World Wide Web Consortium\">W3C</abbr> </a> <sup>®</sup> (<a href=\"https://www.csail.mit.edu/\"> <abbr title=\"Massachusetts Institute of Technology\">MIT</abbr> </a>, <a href=\"https://www.ercim.eu/\"> <abbr title=\"European Research Consortium for Informatics and Mathematics\">ERCIM</abbr> </a>, <a href=\"https://www.keio.ac.jp/\">Keio</a>, <a href=\"https://ev.buaa.edu.cn/\">Beihang</a>). W3C <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\">liability</a>, <a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\">trademark</a> and <a href=\"https://www.w3.org/Consortium/Legal/copyright-software\">permissive document license</a> rules apply. </blockquote> </div><div class=\"source\"> <span style=\"font-style: italic\">Include this source code</span>:<br><code>&lt;p class=\"copyright\"&gt;&lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Copyright\"&gt;Copyright&lt;/a&gt; © 2022&lt;a href=\"https://www.w3.org/\"&gt;&lt;abbr title=\"World Wide Web Consortium\"&gt;W3C&lt;/abbr&gt;&lt;/a&gt;&lt;sup&gt;®&lt;/sup&gt; (&lt;a href=\"https://www.csail.mit.edu/\"&gt;&lt;abbr title=\"Massachusetts Institute of Technology\"&gt;MIT&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.ercim.eu/\"&gt;&lt;abbr title=\"European Research Consortium for Informatics and Mathematics\"&gt;ERCIM&lt;/abbr&gt;&lt;/a&gt;, &lt;a href=\"https://www.keio.ac.jp/\"&gt;Keio&lt;/a&gt;, &lt;a href=\"https://ev.buaa.edu.cn/\"&gt;Beihang&lt;/a&gt;). W3C &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer\"&gt;liability&lt;/a&gt;, &lt;a href=\"https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks\"&gt;trademark&lt;/a&gt; and &lt;a href=\"https://www.w3.org/Consortium/Legal/copyright-software\"&gt;permissive document license&lt;/a&gt; rules apply.&lt;/p&gt;</code> </div></li></ol> </li></ul>",
```

File: lib/rules/headers/copyright.js [hunk 4]
```diff
@@ -37,120 +48,165 @@ const linksToCheckBase = {
...
+    }
+
+    // licenseTexts: ['permissive document license'] or ['document use'] or ['document use', 'permissive document license']
+    const licenseTexts = allowedLicenses
+        .filter(v => [LICENSE_CD_URL, LICENSE_CS_URL].includes(v))
+        .map(v => LICENSE_URL_TEXT_MAP[v]);
+
+    sr.transition({
+        from: new Date(2023, 0, 1),
+        to: new Date(2023, 1, 1),
```
Comment (deniak) [inline lib/rules/headers/copyright.js:204]: The dates comparison in `transition` do not include the limits
```suggestion
        to: new Date(2023, 0, 31),
```

**Review by deniak — CHANGES_REQUESTED** [review]

File: lib/l10n.js (hunk 2 above)
Comment (jennyliang220) [reply lib/l10n.js:152]: resolved by https://github.com/w3c/specberus/issues/1686

**Review by deniak — APPROVED** [review]

**Pull request merged**